### ISSUE TITLE
There is no env variable to specify additional JDBC connection string options for application DB

### ISSUE THREAD

**Issue by X-Stranger** [metabase/metabase#5750]
We need to set ```prepareThreshold=0``` as additional parameter to Postgres JDBC connection string when run Metabase in Docker container. There are env variables for db type, name, port, user, password, hostname. But there is no env variable for additional options.

I propose to create MB_DB_OPTIONS variable and add support for in in source code:
```diff
diff --git a/src/metabase/db.clj b/src/metabase/db.clj
index 464d6b0ff..ba071c7d3 100644
--- a/src/metabase/db.clj
+++ b/src/metabase/db.clj
@@ -86,7 +86,8 @@
                           :port     (config/config-int :mb-db-port)
                           :dbname   (config/config-str :mb-db-dbname)
                           :user     (config/config-str :mb-db-user)
-                          :password (config/config-str :mb-db-pass)}))))
+                          :password (config/config-str :mb-db-pass)
+                          :additional-options (config/config-str :mb-db-options)}))))
 
 (defn jdbc-details
   "Takes our own MB details map and formats them properly for connection details for JDBC."
```

:arrow_down: Please click the :+1: reaction instead of leaving a `+1` or `update?` comment

**Comment by krainboltgreene** [comment]
This is *required* to use pgbouncer/pgpool in certain modes.

**Comment by flamber** [comment]
I think the was fixed in 0.31, when `MB_DB_CONNECTION_URI` was added as env variable, which allows you to use a full JDBC URI? https://www.metabase.com/docs/latest/operations-guide/start.html

**Comment by krainboltgreene** [comment]
Sadly I can't use the uri because I have a password as a secret.

**Comment by flamber** [comment]
@krainboltgreene 
I don't think I understand. So you connect to your Postgres via pgbouncer, which uses a vault to get the password? Have you tried not passing the `&password=...` and just letting pgbouncer do what it normally does?
Or does your setup break the way JDBC works with Postgres?
https://jdbc.postgresql.org/documentation/head/connect.html

**Comment by krainboltgreene** [comment]
To be clear: metabase connects (with a username/password) to pgbouncer which load balances to postgresql.

  0. The password is available to the container as a ENV via a Kubernetes Secret
  0. https://pgbouncer.github.io/faq.html#how-to-use-prepared-statements-with-transaction-pooling

I could just share the URI via a kubernetes secret however:

  0. If I need the password alone (I do) then I need to keep two secrets in sync, one with the password and one with the uri(that contains the password)
  0. It'd be the only postgres-connecting container to do it this way in my ecosystem

Trust me, I love uris for services, but in my ecosystem it is advantageous to have the password as a singular Secret that I can pass around to containers, which means it's disadvantageous to require me to use the URI form.

**Comment by camsaul** [comment]
I think one thing we could **possibly** do is support `MB_DB_USER` and `MB_DB_PASS` in combination with `MB_DB_CONNECTION_URI`. That's mostly straightforward once #19970 is merged in since `java.sql.DriverManager` supports that natively. I don't think we'd ever add support for an "additional options" env var because there is exactly one person ever who has asked for this.

**Comment by camsaul** [comment]
Opened #20122 to track using `MB_DB_USER` and `MB_DB_PASS` in combination with `MB_DB_CONNECTION_URI`

**Comment by camsaul** [comment]
Closing as Won't Fix. In Metabase 0.43.0 you can use `MB_DB_CONNECTION_URI` (in combination with `MB_DB_PASS` and `MB_DB_USER` if needed) to set any options you want
